Tests/SonarScanner.MSBuild.Tasks.UnitTests/IsTestByReferenceTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarScanner.MSBuild.Tasks.UnitTests
+{
+    [TestClass]
+    public class IsTestByReferenceTests
+    {
+        [DataTestMethod]
+        [DataRow("SimpleReference")]
+        [DataRow("mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+        [DataRow("mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089", "StrongNameAndSimple")]
+        public void TestReference_ProductReference_IsNull(params string[] references) =>
+            ExecuteAndAssert(references, null, "Resolved test reference: ");
```

Comment:
here I find the log confusing. IMO if it's null, the message should be "No test reference found."